specification/kubernetesruntime/KubernetesRuntime.Management/storageclass.tsp

```diff
@@ -0,0 +1,276 @@
+enum PerformanceTier {
+  @doc("Undefined Performance Tier")
+  Undefined,
+
+  @doc("Basic Performance Tier")
+  Basic,
+
+  @doc("Standard Performance Tier")
+  Standard,
+
+  @doc("Premium Performance Tier")
+  Premium,
+
+  @doc("Ultra Performance Tier")
+  Ultra,
+}
+
+@doc("Failover tier of a storage class")
+enum FailoverTier {
+  @doc("Not available Failover Tier")
+  NotAvailable,
+
+  @doc("Slow Failover Tier")
+  Slow,
+
+  @doc("Fast Failover Tier")
+  Fast,
+
+  @doc("Super Failover Tier")
+  Super,
+}
+
+@doc("Type of a storage class")
+enum SCType {
+  @doc("Native storage class")
+  Native,
+
+  @doc("RWX storage class")
+  RWX,
+
+  @doc("Blob storage class")
+  Blob,
+
+  @doc("NFS storage class")
+  NFS,
+
+  @doc("SMB storage class")
+  SMB,
+}
+
+@doc("Data resilience tier of a storage class")
+enum DataResilienceTier {
+  @doc("Not data resilient")
+  NotDataResilient,
+
+  @doc("Data resilient")
+  DataResilient,
+}
+
+@doc("Ability to expand volumes of a storage class")
+enum VolumeExpansion {
+  @doc("Allow volume expansion")
+  Allow,
+
+  @doc("Disallow volume expansion")
+  Disallow,
+}
+
+@doc("Storage Class Access Mode")
+enum AccessMode {
+  @doc("Read Write Once (RWO) access mode")
+  ReadWriteOnce,
+
+  @doc("Read Write Many (RWX) access mode")
+  ReadWriteMany,
+}
+
+@doc("Storage class volume binding mode")
+enum VolumeBindingMode {
+  @doc("Immediate binding mode")
+  Immediate,
+
+  @doc("Wait for first consumer binding mode")
+  WaitForFirstConsumer,
+}
+
+@doc("Details of the StorageClass StorageClass.")
+model StorageClassProperties {
+  @doc("Volume can be expanded or not")
+  allowVolumeExpansion?: VolumeExpansion;
+
+  @doc("Additional mount options")
+  mountOptions?: string[];
+
+  @doc("Provisioner name")
+  @visibility("read", "create")
+  provisioner?: string;
+
+  @doc("Binding mode of volumes: Immediate, WaitForFirstConsumer")
+  @visibility("read", "create")
+  volumeBindingMode?: VolumeBindingMode;
+
+  @doc("The access mode: [ReadWriteOnce, ReadWriteMany] or [ReadWriteOnce]")
+  accessModes?: AccessMode[];
+
+  @doc("Allow single data node failure")
+  dataResilience?: DataResilienceTier;
+
+  @doc("Failover speed: NA, Slow, Fast")
+  failoverSpeed?: FailoverTier;
+
+  @doc("Limitations of the storage class")
+  limitations?: string[];
+
+  @doc("Performance tier")
+  performance?: PerformanceTier;
+
+  @doc("Selection priority when multiple storage classes meet the criteria. 0: Highest, -1: Never use")
+  priority?: int64;
+
+  @doc("Properties of the StorageClass")
+  // prevent generation of buggy StorageClassPropertiesUpdate model
+  @visibility("read", "create")
+  typeProperties: StorageClassTypeProperties;
+
+  @doc("Resource provision state")
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+}
+
+@discriminator("type")
+@doc("The properties of storage class of the StorageClass")
+model StorageClassTypeProperties {
+  @doc("Type of the storage class.")
+  type: SCType;
```

Comment:
should these be @visibility("read", "create")?

Reply:
`StorageClassTypeProperties` is only used in put model, not in `PATCH` model. The patch model uses `StorageClassTypePropertiesUpdate` for `typeProperties` which doesn't contain `type` whatsoever. So it doesn't matter if this field has `@visibility` or not.